### PR TITLE
add support in `.fetch_results` for retrieving results in multiple pages

### PR DESCRIPTION
**Pull request by benubah**
This change improves the existing code based on the unexpected behavior reported in the following issue  https://github.com/rladies/meetupr/issues/55

### CODE REVIEW

File: R/internals.R [hunk 1]
```diff
@@ -31,45 +33,42 @@
...
-
-      next_url <- strsplit(strsplit(res$headers$link, split = "<")[[1]][2], split = ">")[[1]][1]
-      res <- .quick_fetch(next_url, api_key, event_status)
-      all_records[[i + 1]] <- res$result
-    }
-    records <- unlist(all_records, recursive = FALSE)
-  }
-
+  
+    if((length(records) < total_records) & !is.null(res$headers$link)){
```
Comment (ledell) [inline R/internals.R:58]:
```suggestion
    if ((length(records) < total_records) & !is.null(res$headers$link)) {   
```

File: R/internals.R [hunk 2]
```diff
@@ -31,45 +33,42 @@
...
-    }
-    records <- unlist(all_records, recursive = FALSE)
-  }
-
+  
+    if((length(records) < total_records) & !is.null(res$headers$link)){   
+      # calculate number of offsets for records above 200
+      offsetn <- ceiling(total_records/length(records))
+      all_records <- list(records)     
+      for(i in 1:(offsetn - 1)) {
```
Comment (ledell) [inline R/internals.R:62]:
```suggestion
      for (i in 1:(offsetn - 1)) {
```

**Comment by ledell** [comment]
Hi @benubah I support the changes, except that I got an error when running the code above.  Since I needed the updated `find_groups()` function from https://github.com/rladies/meetupr/pull/48 to try the code example, I copy-pasted the function in to my shell, so this might have caused an issue, but I ran the code and got an error.  Can you double check that this works?

```
> all_ruser_groups <- find_groups(text = "r-project-for-statistical-computing", fields = "past_event_count, upcoming_event_count, last_event, topics", api_key = meetup_api_key)
Downloading 482 record(s)...Error in as_mapper(.f, ...) : argument ".f" is missing, with no default
```  

I think it makes sense to add these changes in the other PR since they are related to the other PR and required to make that one fully functional (and also makes it easier to test).

**Comment by benubah** [comment]
@ledell Please see a screenshot of my terminal. It works smoothly for me.
The PR updates `.quick_fetch()` and `.fetch_results()`. If you look at the full git diff https://github.com/rladies/meetupr/pull/57/commits/ec0943b3753431d5de986ab9f4f14b8e55ddf470   for `internals.R` from start to finish, few insertions of `offset` as an argument were made. See what `.quick_fetch()` now looks like below:
But, I have now added the changes to the other PR https://github.com/rladies/meetupr/pull/48 to make testing easier.

I did not attach the `meetupr` package. I just sourced the improved `internals.R` and `find_groups()`

```r
.quick_fetch <- function(api_url,
                         api_key = NULL,
                         event_status = NULL,
                         offset = 0,
                         ...) 
```

image

Feel free to let me know if it still does not work.

**Comment by ledell** [comment]
@benubah Thanks, I assume the error I got had to do with me copy/pasting code.  I'll try it out on the other PR...

**Comment by ledell** [comment]
Hi @benubah can you fix the conflict so we can get this merged?  Sorry again for not merging this last year.

**Comment by maelle** [comment]
:wave: @benubah

**Comment by maelle** [comment]
Given the changes that were made in the default branch, and given that some functions already handle pagination, I think it might be easier to start from scratch/ using code from these functions.

**Pull request closed without merging**